custom_components/google_home/models.py

```diff
@@ -0,0 +1,137 @@
+from datetime import timedelta
+import json
+from typing import Optional
+
+from homeassistant.util.dt import as_local, utc_from_timestamp
+
+from .const import (
+    DATETIME_STR_FORMAT,
+    DOMAIN,
+    FIRE_TIME,
+    ID,
+    LABEL,
+    ORIGINAL_DURATION,
+    RECURRENCE,
+)
+
+
+def convert_from_ms_to_s(timestamp):
+    return round(timestamp / 1000)
+
+
+class GoogleHomeDevice:
+    """Local representation of Google Home device"""
+
+    def __init__(
+        self,
+        name: str,
+        token: str,
+        ip_address: Optional[str] = None,
+        hardware: Optional[str] = None,
+    ) -> None:
+        self.name = name
+        self.token = token
+        self.ip_address = ip_address
+        self.hardware = hardware
+        self.available = True
+        self._timers = []
+        self._alarms = []
+        self.integration = DOMAIN
+
+    def set_alarms(self, alarms):
+        """Stores alarms as GoogleHomeAlarm objects"""
+        self._alarms = [
+            GoogleHomeAlarm(
+                _id=alarm[ID],
+                fire_time=alarm[FIRE_TIME],
+                label=alarm.get(LABEL),
+                recurrence=alarm.get(RECURRENCE),
+            )
+            for alarm in alarms
+        ]
+
+    def set_timers(self, timers):
+        """Stores timers as GoogleHomeTimer objects"""
+        self._timers = [
+            GoogleHomeTimer(
+                _id=timer[ID],
+                fire_time=timer[FIRE_TIME],
+                duration=timer[ORIGINAL_DURATION],
+                label=timer.get(LABEL),
+            )
+            for timer in timers
+        ]
+
+    def get_alarms(self):
```

Comment:
How about this? It will be cleaner in places there this is used.
```suggestion
    def get_sorted_alarms(self):
```